### PR TITLE
create a sponser react component

### PR DESCRIPTION
**Pull request by Mohmmed-Mahsoub**

### CODE REVIEW

File: src/components/sponsers/index.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+//import { charityAPI } from '../../clients';
+import sponserImg from './img/envato-dark.png';
+
+class Responsers extends React.Component {
```
Comment (Nilegfx) [inline src/components/sponsers/index.jsx:5]: Sponsers not Responsers

File: src/App.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { charityAPI } from './clients';
 import Logo from './components/Logo';
 import MainContact from './components/MainContact';
 import FeaturedBanner from './components/FeaturedBanner';
+import Sponsers from './components/sponsers';
```
Comment (Nilegfx) [inline src/App.js:6]: As a convention, We name React component files (or the directory that contains the file) capitalized. In your case, `components/Sponser`

File: src/components/sponsers/index.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  render() {
+    return (
+      <div className="container">
+        <div className="flex justify-between mx-auto my-32 flex-wrap">
+          <div className="sponser">
```
Comment (Nilegfx) [inline src/components/sponsers/index.jsx:10]: Get Sponsers from the state, loop. "Draw them dynamically"

File: src/components/sponsers/index.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { charityAPI } from '../../clients';
+import './styles.css';
+
+class Sponsers extends React.Component {
+  state = { urls: [], urlsError: false };
+  componentDidMount() {
+    charityAPI('/Sponsers')
+      .then(res => {
+        console.log(res);
```
Comment (elhaw) [inline src/components/sponsers/index.jsx:10]: you need to remove` console.log`

File: src/components/sponsers/index.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { charityAPI } from '../../clients';
+import './styles.css';
+
+class Sponsers extends React.Component {
+  state = { urls: [], urlsError: false };
```
Comment (elhaw) [inline src/components/sponsers/index.jsx:6]: Add `loading` to state  set to `true`

File: src/components/sponsers/index.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  componentDidMount() {
+    charityAPI('/Sponsers')
+      .then(res => {
+        console.log(res);
+        this.setState({
+          urls: res.data
+        });
```
Comment (elhaw) [inline src/components/sponsers/index.jsx:13]: After we received the response you changed the state of `urls`.
You also need to change the state of `error `and `loading`.
_whether you received the **data you expect** or **error** it means that no more loading_ so you need to set the state of` loading: false` and `error :false`

File: src/components/sponsers/index.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      })
+      .catch(error => {});
```
Comment (elhaw) [inline src/components/sponsers/index.jsx:15]: Here also you need to handle state change of `error`, `loading`

```
      .catch(error => {
        this.setState({
         //change state of error to true 
         //change state of loading to false
        });
      });
```

Comment (AbdoMgdy) [inline src/components/sponsers/index.jsx:15]: you need to display an error message here that can be rendered on the page .

File: src/components/sponsers/index.jsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  render() {
+    return this.state.urls.map(({ _id, image }) => (
```
Comment (elhaw) [inline src/components/sponsers/index.jsx:18]: This function will work if you received the data you expect but. 
But what **_if data is  still loading_** (We need to tell the user that something is loading) or what if some error occurs (We need to handle that error )
So you need to add some conditional statment
```
    if (//error is true) {
      return <div className="error">error here</div>;
    }

    if (// still loading is true) {
      return <div className="loading">loading .. </div>;
    }
```

**Pull request merged**